msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/TokenSource.java

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+/**
+ * A list of possible sources for the tokens found in an {@link IAuthenticationResult}
+ */
+public enum TokenSource {
+
+    /**
+     * A default value, likely indicates tokens could not be retrieved
+     */
+    UNKNOWN,
```

Comment:
MSAL throws exception when token cannot be retried. I don't think you can get this value. Also, there is no unit test covering this path.